contrib/check-config.sh

```diff
@@ -121,6 +121,9 @@ check_device() {
 }
 
 check_distro_userns() {
+	if [ ! -e /etc/os-release ]; then
+		return
+	fi
 	. /etc/os-release 2> /dev/null || /bin/true
```

Comment:
I was chatting with @tianon as I expected this line would be able to handle this situation, but apparently "it depends" (some shells may fail on this, others not)

Reply:
Doesn't OpenWrt just use `busybox` for `/bin/sh`?  I would've thought it would be well-behaved here, but apparently I'm wrong: :weary:

```console
$ docker run --rm busybox sh -c 'rm /etc/os-release; . /etc/os-release || echo successful fallback'
rm: can't remove '/etc/os-release': No such file or directory
sh: .: line 1: can't open '/etc/os-release': No such file or directory
```

According to POSIX, the behavior here is technically undefined: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#dot (it only says what should happen in the case of an interactive shell or if the file is successfully read :disappointed:).

Reply:
Realistically, we have more changes we need to make if we want this script to actually be friendly to a pure BusyBox environment: :grimacing:

```console
$ docker run -it --rm busybox
/ # wget -q 'https://github.com/moby/moby/raw/master/contrib/check-config.sh'
wget: note: TLS certificate validation not implemented
/ # echo 'CONFIG_USER_NS=y' > .config
/ # CONFIG=.config sh check-config.sh
info: reading kernel config from .config ...

Generally Necessary:
- cgroup hierarchy: cgroupv2
  Controllers:
  - cpu: available
  - cpuset: available
  - io: available
  - memory: available
  - pids: available
- apparmor: enabled, but apparmor_parser missing
    (look for an "apparmor" package for your distribution)
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_NAMESPACES: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_NET_NS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_PID_NS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_IPC_NS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_UTS_NS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_CGROUPS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_CGROUP_CPUACCT: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_CGROUP_DEVICE: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_CGROUP_FREEZER: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_CGROUP_SCHED: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_CPUSETS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_MEMCG: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_KEYS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_VETH: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_BRIDGE: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_BRIDGE_NETFILTER: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_IP_NF_FILTER: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_IP_NF_TARGET_MASQUERADE: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_NETFILTER_XT_MATCH_CONNTRACK: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_NETFILTER_XT_MATCH_IPVS: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_NETFILTER_XT_MARK: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_IP_NF_NAT: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_NF_NAT: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_POSIX_MQUEUE: missing
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_CGROUP_BPF: missing

Optional Features:
- zcat: no gzip/bzip2/xz magic
zcat: no gzip/bzip2/xz magic
CONFIG_USER_NS: missing
```